src/infer_single_paired.py

```diff
@@ -0,0 +1,262 @@
+"""Infers single or paired end library for a given file.
+
+This module parses fastq files and depending on whether they are
+first mate or second mate, it returns tuple of values accordingly.
+Tuple description :
+    First member of the tuple corresponds to the result for the
+    first file with possible values- 0,1,2,3,None.
+    (None) if no file provided.
+    Second member of the tuple corresponds to the result for the
+    second file with possible values- 0,1,2,3,None.
+    (None) if no file provided.
+    Third member of the tuple indicates 4 if the files have split paired
+    end library and 5 if the read id's don't match among the two files.
+
+The modules parses sucessfully for fastq files that are strictly
+as per as wiki convention.
+
+Examples:
+    @HWUSI-EAS100R:6:73:941:1973#0/1
+    @EAS139:136:FC706VJ:2:2104:15343:197393 1:Y:18:ATCACG
+
+"""
+import sys
+import gzip
+import re
+from functools import partial
+from enum import Enum
+from Bio.SeqIO.QualityIO import FastqGeneralIterator
+sys.tracebacklimit = 0
+
+
+class Messages(Enum):
+    invalid_file = 0
+    first_mate = 1
+    second_mate = 2
+    mixed_paired_end = 3
+    split_paired_end = 4
+    different_read_ids = 5
+
+
+class EndParser:
+    def __init__(self, file1_name: str, file2_name: str):
+        """Constructor to assign filenames and initialize file pointers.
+
+        Args:
+            file1_name (str) : First file name provided by the user.
+            file2_name (str) : Second file name provide by the user.
+
+        Attributes:
```

Comment:
You have 4 attributes in this case, two file names and two file pointers; for the first two of them, you should copy the description from section "Args:".